nntrainer/dataset/raw_file_data_producer.cpp

```diff
@@ -0,0 +1,180 @@
+
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2021 Jihoon Lee <jhoon.it.lee@samsung.com>
+ *
+ * @file   raw_file_data_producer.cpp
+ * @date   12 July 2021
+ * @brief  This file contains raw file data producers, reading from a file
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Jihoon Lee <jhoon.it.lee@samsung.com>
+ * @bug    No known bugs except for NYI items
+ *
+ */
+
+#include <raw_file_data_producer.h>
+
+#include <fstream>
+#include <memory>
+#include <numeric>
+#include <random>
+#include <vector>
+
+#include <base_properties.h>
+#include <nntrainer_error.h>
+#include <node_exporter.h>
+#include <util_func.h>
+
+namespace nntrainer {
+
+/**
+ * @brief Props containing file path value
+ *
+ */
+class PropsPath : public Property<std::string> {
+public:
```

Comment:
Isn't it better to collect all the property in one location? like in common_property?

Reply:
Okay :) I'll make this to public.

Reply:
> AFIK, there will be a specific directory name coming up after this file dataset. In that case, what the type name might be if it is path? how about filepath, dirpath for making it clearer?

One concern here is that, other's using this property need different validation logic. In this case it might be better to put it here, I think we can consider this later so I'll put this inside common properties :)

Reply:
Done